src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,284 @@
+import React, {
+  KeyboardEvent,
+  FocusEvent,
+  MouseEvent,
+  useState,
+  useEffect,
+  useRef,
+} from 'react'
+import classnames from 'classnames'
+
+interface ComboBoxOption {
+  value: string
+  label?: string
+}
+
+enum Direction {
+  Up = -1,
+  Down = 1,
+}
+
+enum FocusMode {
+  None,
+  Input, // The textfield
+  Item, // One of the list items
+}
+
+interface ComboBoxProps {
+  id: string
+  name: string
+  className?: string
+  options: ComboBoxOption[]
```

Comment:
I could be missing something but how would we load this component with an option already selected?  For example, of we are displaying form data coming back from an api. In my mind there has to be some kind way to set the selected item from outside the component (whether that's tracking selected state within the `ComboBoxOption`, or having a separate `selectedOption` prop, or using the current input value to select an option on initial load).

Reply:
This was missing. I added support for passing in a `defaultValue` prop, which I think will satisfy this and I think is the way you'd do it with a standard `select` in React? There is a story that covers this case now, but not a unit test yet.

Reply:
Oh that makes sense!